src/ring_theory/norm.lean

```diff
@@ -209,10 +243,34 @@ begin
   { intro σ, rw [power_basis.lift_equiv'_apply_coe, id.def] }
 end
 
+lemma norm_eq_prod_roots [is_separable K L] [finite_dimensional K L]
+  {x : L} (hF : splits (algebra_map K F) (minpoly K x)) :
+  (algebra_map K F) (norm K x) =
+  ((minpoly K x).map (algebra_map K F)).roots.prod ^ finrank (K⟮x⟯) L :=
```

Comment:
I don't like that these statements don't fit on 1 line anymore. I've been seeing similar problems in `algebraic_geometry` recently.
Can we take a small step back, and see if we can improve this?

Quick ideas:

* `algebra_map` is a long name for a common function. See also this discussion https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/notation.20for.20algebra_map
* `finrank` is also pretty long. Should this just be `rk`? With a longer name for "rank as cardinal".
* `(K⟮x⟯)`, with proper precendence the outer `()` shouldn't be needed, I hope.

Reply:
I agree that `algebra_map` is a long name for such a common map. I am not sure if we use more `finrank` or `rank`.

In any case I've golfed the statement a little, it now fits in one line. :) (The outer `()` around `K⟮x⟯` was already unnecessary.)

Reply:
I count:
 * 966 instances of "finrank" as a separate word
 * 386 instances of "rank" as a separate word
but also
 * 642 instances of "dim" as a separate word.

So I would estimate that rank/dimension as a cardinality is about as popular as rank as a natural number. While dropping a few letters from "finrank" would fix this line, it wouldn't help the other parts that are quite long compared to typical mathematical notation, such as `(minpoly K x).map (algebra_map K F)`. If we instead used coercions, we'd get `(norm K (adjoin_simple.gen K x) : F) = (minpoly K x : polynomial F).roots.prod` which is a lot better.